lib/preload.js

```diff
@@ -242,10 +319,172 @@ class Preloader extends EventEmitter {
     })
   }
 
+  _prepareErrorHandler () {
+    // Emit an error event if the python script exits with an error
+    this.container.wait()
+    .then((data) => {
+      if ((data.StatusCode !== 0) && (data.StatusCode !== EXIT_CAUSED_BY_SIGINT)) {
+        const output = this.bufferedStderr.getData().toString('utf8').trim()
+        let error
+        if ((output.indexOf(GRAPHDRIVER_ERROR) !== -1) && (output.indexOf(DOCKERD_USES_OVERLAY) !== -1)) {
+          error = new errors.ResinError(OVERLAY_MODULE_MESSAGE)
+        } else {
+          error = new Error(output)
+          error.code = data.StatusCode
+        }
+        this.emit('error', error)
+      }
+    })
+  }
+
+  _runCommand (command, parameters) {
+    return new Promise((resolve, reject) => {
+      this.stdout.once('error', reject)
+      this.stdout.once('data', (data) => {
```

Comment:
Are you sure `once` is enough to gather all output? It seems more appropriate to register with `.on` and remove the listener when `data === null`.

Reply:
`once` enough here. The responses are always JSON encode strings with no newlines.
I'd prefer not changing this in this PR.